chain/chaintest/block.go

```diff
@@ -104,3 +111,270 @@ func GenerateTestExecutedBlocks(
+				return nil, err
+			}
+
+			maxFee, err := fees.MulSum(unitPrices, units)
+			if err != nil {
+				return nil, err
+			}
+
+			txData := chain.NewTxData(
+				chain.Base{
+					Timestamp: utils.UnixRMilli(timestamp, rules.GetValidityWindow()),
+					ChainID:   rules.GetChainID(),
+					MaxFee:    maxFee,
+				},
+				actions,
+			)
+
+			return txData.Sign(factory)
+		}
+
+		txs, err := txListGenerator(numOfTxsPerBlock, txGenerator)
+		if err != nil {
+			return nil, err
+		}
+
+		ts := tstate.New(0)
+		for _, tx := range txs {
+			stateKeys, err := tx.StateKeys(balanceHandler)
+			if err != nil {
+				return nil, err
+			}
+			tsv := ts.NewView(stateKeys, parentCtx.view, 0)
+			if err := tx.PreExecute(ctx, feeManager, balanceHandler, rules, tsv, timestamp); err != nil {
+				return nil, err
+			}
+			result, err := tx.Execute(ctx, feeManager, balanceHandler, rules, tsv, timestamp)
+			if err != nil {
+				return nil, err
+			}
+			ok, d := feeManager.Consume(result.Units, rules.GetMaxBlockUnits())
+			if !ok {
+				return nil, fmt.Errorf("consumed too many units in dim %d", d)
+			}
+			tsv.Commit()
+		}
+
+		tsv := ts.NewView(state.CompletePermissions, parentCtx.view, 0)
+		if err := tsv.Insert(ctx, chain.HeightKey(metadataManager.HeightPrefix()), binary.BigEndian.AppendUint64(nil, height)); err != nil {
+			return nil, err
+		}
+		if err := tsv.Insert(ctx, chain.TimestampKey(metadataManager.TimestampPrefix()), binary.BigEndian.AppendUint64(nil, uint64(timestamp))); err != nil {
+			return nil, err
+		}
+		if err := tsv.Insert(ctx, chain.FeeKey(metadataManager.FeePrefix()), feeManager.Bytes()); err != nil {
+			return nil, err
+		}
+		tsv.Commit()
+
+		root, err := parentCtx.view.GetMerkleRoot(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		parentCtx.view, err = parentCtx.view.NewView(ctx, merkledb.ViewChanges{
+			MapOps:       ts.ChangedKeys(),
+			ConsumeBytes: true,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		statelessBlock, err := chain.NewStatelessBlock(
+			parentCtx.id,
+			timestamp,
+			height,
+			txs,
+			root,
+			nil,
+		)
+		if err != nil {
+			return nil, err
+		}
+		parentCtx.id = statelessBlock.GetID()
+
+		blk := chain.NewExecutionBlock(statelessBlock)
+		executionBlocks[i] = blk
+	}
+	return executionBlocks, nil
+}
+
+// BlockBenchmark is a parameterized benchmark. It generates [NumOfBlocks] blocks
+// with [NumOfTxsPerBlock] transactions per block, and then calls
+// Processor.Execute to process the block list [b.N] times.
+type BlockBenchmark struct {
+	MetadataManager chain.MetadataManager
+	BalanceHandler  chain.BalanceHandler
+	RuleFactory     chain.RuleFactory
+	AuthEngines     chain.AuthEngines
+
+	BlockBenchmarkHelper BlockBenchmarkHelper
+
+	Config                chain.Config
+	AuthVerificationCores int
+
+	// [NumOfBlocks] is set as a hyperparameter to avoid using [b.N] to generate
+	// the number of blocks to run the benchmark on.
+	NumOfBlocks      uint64
+	NumOfTxsPerBlock uint64
+}
```

Comment:
```suggestion
	Config                chain.Config
	AuthVerificationCores int
	
	BlockBenchmarkHelper BlockBenchmarkHelper

	// [NumOfBlocks] is set as a hyperparameter to avoid using [b.N] to generate
	// the number of blocks to run the benchmark on.
	NumOfBlocks      uint64
	NumOfTxsPerBlock uint64
}
```